kong/runloop/balancer/round_robin.lua

```diff
@@ -92,36 +97,51 @@ function roundrobin_algorithm:getPeer(cacheOnly, handle, hashValue)
     end
 
     address = self.wheel[self.pointer]
-    if address ~= nil and address.available and not address.disabled then
-      ip, port, hostname = balancers.getAddressPeer(address, cacheOnly)
-      if ip then
-        -- success, update handle
-        handle.address = address
-        return ip, port, hostname, handle
-
-      elseif port == balancers.errors.ERR_DNS_UPDATED then
-        -- if healty we just need to try again
-        if not self.balancer.healthy then
-          return nil, balancers.errors.ERR_BALANCER_UNHEALTHY
-        end
-      elseif port == balancers.errors.ERR_ADDRESS_UNAVAILABLE then
-        ngx.log(ngx.DEBUG, "found address but it was unavailable. ",
-          " trying next one.")
-      else
-        -- an unknown error occurred
-        return nil, port
+    if not address or not address.available or address.disabled then
+      goto continue
+    end
+
+    -- address ~= nil and address.available and not address.disabled
+
+    ip, port, hostname = balancers.getAddressPeer(address, cacheOnly)
+
+    if ip then
+      -- success, update handle
+      handle.address = address
+      return ip, port, hostname, handle
+    end
+
+    -- no ip, port is an error hint
+
+    if port == balancers.errors.ERR_DNS_UPDATED then
+      -- if healthy we just need to try again
+      if self.balancer.healthy then
+        goto continue
       end
 
+      -- not healthy
+      return nil, balancers.errors.ERR_BALANCER_UNHEALTHY
     end
 
+    if port == balancers.errors.ERR_ADDRESS_UNAVAILABLE then
+      ngx.log(ngx.DEBUG, "found address but it was unavailable. ",
+                         " trying next one.")
+      goto continue
+    end
+
+    -- an unknown error occurred
+    do return nil, port end
+
+    ::continue::
+
   until self.pointer == starting_pointer
 
   return nil, balancers.errors.ERR_NO_PEERS_AVAILABLE
 end
 
 
 function roundrobin_algorithm.new(opts)
-  assert(type(opts) == "table", "Expected an options table, but got: "..type(opts))
+  assert(type(opts) == "table", "Expected an options table, but got: " .. type(opts))
```

Comment:
For this style change, I think we need to let the lint or else tool do it, we don't need to do this in every PR.

Reply:
Agree, but now we don't have such tool.